vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -47,16 +49,66 @@ type InstallerController struct {
 
 	// installerPodImageFn returns the image name for the installer pod
 	installerPodImageFn func() string
+
+	// installStateFunc derives the static pod install state of a node
+	staticPodStateFunc StaticPodStateFunc
+}
+
+// StaticPodState is the status of a static pod that has been installed to a node.
+type StaticPodState int
+
+const (
+	// staticPodStatePending means that the installed static pod is not up yet.
+	StaticPodStatePending = StaticPodState(iota)
+	// StaticPodStateReady means that the installed static pod is ready.
+	StaticPodStateReady
+	// StaticPodStateFailed means that the static pod installation of a node has failed.
+	StaticPodStateFailed
+)
+
+const deploymentIDLabel = "deployment-id"
+
+// StaticPodStateFunc derives the status of the installed static pod. It returns the state, an array of error strings
+// in case of StaticPodStateFailed and an error in case the state couldn't be checked.
+type StaticPodStateFunc func(targetDeploymentID int32, nodeName string) (StaticPodState, []string, error)
+
+// NewDefaultStaticPodStateFunc returns a static pod state func which checks the pods readiness via the pod condition.
+func NewDefaultStaticPodStateFunc(podlister corev1listers.PodLister, ns string, staticPodNamePrefix string) StaticPodStateFunc {
+	return func(deploymentID int32, nodeName string) (StaticPodState, []string, error) {
+		pod, err := podlister.Pods(ns).Get(staticPodNamePrefix + "-" + nodeName)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return StaticPodStatePending, nil, nil
+			}
+			return StaticPodStatePending, nil, err
+		}
+		if pod.Labels[deploymentIDLabel] != strconv.Itoa(int(deploymentID)) {
```

Comment:
if we get a pod which does not have the matching revision (deployment id) what does it mean for this controller? how can that happen? do we want to ignore that pod and wait for the deployment ID passed to this function?

Reply:
It happens all the time during deployment. Until the kubelet has updated the static pod, the controller sees the old deployment id.

Reply:
This `deploymentID` should be named `targetDeploymentID`. Renamed that now. Then it's clearer I think.

Reply:
done